gitcache/pkg/service.go

```diff
@@ -0,0 +1,190 @@
+package pkg
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/mholt/archiver"
+	"github.com/pkg/errors"
+)
+
+type cacheService struct {
+	BlobURL string
+	Logf    func(s string, f ...interface{})
+}
+
+// CacheService updates git cache.
+type CacheService interface {
+	// UpdateCache updates the git cache into the gocloud.dev blob storage.
+	UpdateCache(string) error
+}
+
+// NewCacheService returns new CacheService.
+func NewCacheService(blobURL string, logf func(s string, f ...interface{})) (CacheService, error) {
+	if blobURL == "" {
+		return nil, errors.New("BLOB_URL env cannot be empty")
+	}
+	if logf == nil {
+		return nil, errors.New("Log function cannot be nil")
+	}
+	return cacheService{
+		BlobURL: blobURL,
+		Logf:    logf,
+	}, nil
+}
+
+// UpdateCache updates the git cache into the gocloud.dev blob storage.
+func (c cacheService) UpdateCache(s string) error {
+	var gitRepo *git.Repository
+	// opening the blob
+	bucket, err := NewBucket(c.BlobURL)
+	if err != nil {
+		return errors.Wrapf(err, "unable to open the bucket %s", c.BlobURL)
+	}
+
+	repo := RepoURL{}
+	if err := repo.Set(s); err != nil {
+		return errors.Wrapf(err, "unable parse the URL %s", s)
+	}
+
+	// gets all the path configuration.
+	storage, err := NewStoragePath(repo)
+	if err != nil {
+		return errors.Wrapf(err, "unable get storage")
+	}
+
+	defer storage.Cleanup()
+
+	alreadyUptoDate := false
+	// checks if there is an existing git repo in the bucket
+	if data, exists := bucket.Get(storage.BlobGitFolderPath); exists {
+		c.Logf("bucket ", c.BlobURL, " already has git folder")
```

Comment:
Since we changed this to Info**f**, all these calls need to be changed to format strings right?

Reply:
Ideally yes, But they still output
```
{"level":"info","ts":1614893576.330445,"caller":"gitcache/main.go:77","msg":"BLOB_URL gs://ossf-scorecards-cache"}
{"level":"info","ts":1614893576.331205,"caller":"gitcache/main.go:82","msg":"Starting server for testing HTTP POST...\n"}
{"level":"info","ts":1614893600.9583843,"caller":"pkg/service.go:65","msg":"bucket %!(EXTRA string=gs://ossf-scorecards-cache, string= already has git folder)"}
{"level":"info","ts":1614893601.059973,"caller":"pkg/service.go:78","msg":"bucketb %!(EXTRA string=gs://ossf-scorecards-cache, string= git folder is already up to date.)"}
{"level":"info","ts":1614893601.1506875,"caller":"pkg/service.go:84","msg":"Finished processing %!(EXTRA pkg.RepoURL={github.com ossf criticality_score})"}
```